packages/frontend/src/main/components/viewer/ViewerBubbles.vue

```diff
@@ -228,45 +241,59 @@ export default {
       this.updateBubbles(false)
     )
 
+    // TODO: this needs to be more intelligent...
     this.updateInterval = window.setInterval(this.sendUpdateAndPrune, 2000)
     window.addEventListener('beforeunload', async () => {
       await this.sendDisconnect()
     })
+
+    this.viewer.on(
```

Comment:
regarding all of the event listeners on 'Viewer' - shouldnt these be cleaned up as well? the Viewer instance is re-used, so it seems that these event listeners will keep piling up similar to the window ones

Reply:
Agreed in principle. For now, this would looks like a larger cleanup I don't have the energy for right now (we'd need to handle viewer events centrally somewhere, a bit like a `useViewerEvent()` from `@vueuse`).